### PR TITLE
Sample sets from nonblocking_sample_method can signal .done()

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

**Comment by arcondello** [comment]
For context, the sample set returned by `ScaleComposite` was always indicated that it was done (see [code](https://github.com/dwavesystems/dimod/blob/dc98404439114f075aa7c0ff979ad30079eab2e4/dimod/sampleset.py#L910)).
This then led to `SampleSet.change_vartype` to resolve eagerly (see [code](https://github.com/dwavesystems/dimod/blob/dc98404439114f075aa7c0ff979ad30079eab2e4/dimod/sampleset.py#L1134)).

File: dimod/reference/composites/scalecomposite.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def sample(self, bqm, scalar=None, bias_range=1, quadratic_range=None,
 
         sampleset = self.child.sample(bqm, **parameters)
 
-        yield  # wait until sampleset is read for the remaining part
+        yield sampleset  # so that SampleSet.done() works
```
Comment (randomir) [inline dimod/reference/composites/scalecomposite.py:138]: How about adding a regression test for this?

**Review by randomir — APPROVED** [review]
LGTM!

File: dimod/reference/composites/scalecomposite.py (hunk 1 above)
Comment (arcondello) [reply dimod/reference/composites/scalecomposite.py:138]: As far as I know the only place we actually used this (so far) was in `ScaleComposite`. Also, the previous behaviour was explicitly to ignore whatever was yielded first. I do have an integration test ready for dwave-system that I'll add once this is deployed.

**Pull request merged**